### PR TITLE
Adds *.gateway.dev

### PR DESCRIPTION
**Pull request by tomboc**
* [x] Description of Organization
* [x] Reason for PSL Inclusion
* [x] DNS verification via dig
* [x] Run Syntax Checker (make test)

* [x] Each domain listed in the PRIVATE section has and shall maintain at least two years remaining on registration.

Description of Organization
====

Organization Website: https://cloud.google.com/api-gateway

Engineer at Google on the Cloud API Gateway team.

Reason for PSL Inclusion
====

Cloud API Gateway hosts fully managed HTTP and gRPC proxies with API policy enforcement features for GCP customers. The proxies receive a default hostname `<gateway ID>.<region code>.gateway.dev`

Our customers are not related to each other and there is no mutual trust between the gateways. By adding an entry to the public suffix list we'd like to prevent our customers from setting cookies on each `<region code>.gateway.dev` subdomain. Our use case is similar to `*.r.appspot.com` already included in the list.

DNS Verification via dig
=======

```
dig +short TXT _psl.gateway.dev
"https://github.com/publicsuffix/list/pull/1117"
```

make test
=========

Make test passed.

```
============================================================================
Testsuite summary for libpsl 0.21.1
============================================================================
# TOTAL: 3
# PASS:  3
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================

============================================================================
Testsuite summary for libpsl 0.21.1
============================================================================
# TOTAL: 5
# PASS:  5
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

### CODE REVIEW

File: public_suffix_list.dat [hunk 1]
```diff
@@ -12000,6 +12000,7 @@ blogspot.vn
 cloudfunctions.net
 cloud.goog
 codespot.com
+*.gateway.dev
```
Comment (fsodic) [inline public_suffix_list.dat:12003]: Registry Expiry Date: 2021-05-28

**Review by fsodic — COMMENTED** [review]
This domain not match new rule where domain must active at least 2 years ahead
@dnsguru

**Comment by dnsguru** [comment]
> This domain not match new rule where domain must active at least 2 years ahead
> @dnsguru

```
 nslookup -type=txt _psl.gateway.dev
Server:         8.8.8.8
Address:        8.8.8.8#53

Non-authoritative answer:
*** Can't find _psl.gateway.dev: No answer
```

Nor have the authority records been placed into the DNS, so it appears this is in-progress, so it will be reviewed at a point in the future.

**Review by dnsguru — CHANGES_REQUESTED** [review]
@tomboc talk to @sleevi pls ;
For the section you are updating, we have been requesting that the entries be sorted by TLD then SLD within, so there's some housekeeping within that section and ordering to manage.  Can you review this with the listed maintainer of that section and order this with the sorting method of TLD then SLD within those, (pardon the pun) Alphabetically.

**Comment by dnsguru** [comment]
This will now require a re-base @tomboc due to #1120, and thank you for organizing the section

**Comment by dnsguru** [comment]
@tomboc can you rebase using the .dat file from after the changes submitted in #1126 to reflect the updated TLD list from ICANN?

**Comment by tomboc** [comment]
Thanks for all the comments.
I've rebased the branch to the current HEAD, as of writing there are no conflicts. DNS change took a bit to roll out, but now I can run the `dig` command with expected result. The domain has been renewed and will not expire until 2023-05-28.

**Pull request merged**